cli/lib/kontena/callbacks/master/deploy/50_authenticate_after_deploy.rb

```diff
@@ -49,10 +49,15 @@ def after
         end
 
         require 'shellwords'
-        cmd = "master login --no-login-info --skip-grid-auto-select --verbose --name #{command.result[:name].shellescape} --code #{command.result[:code].shellescape} #{new_master.url.shellescape}"
+        cmd = [
+          'master', 'login', '--no-login-info' ,'--skip-grid-auto-select', '--verbose',
+          '--name', command.result[:name],
+          '--code', command.result[:code],
+          new_master.url
+        ]
         Retriable.retriable do
           ENV["DEBUG"] && $stderr.puts("Running: #{cmd}")
-          Kontena.run(cmd)
+          raise "retry" unless Kontena.run(cmd, returning: :status).zero?
```

Comment:
Not equivalent because `Kontena.run` will re-raise `SystemExit` in error cases, which is not a `StandardError`, and would thus likely behave differently than a `RuntimeError` from `raise "retry"`.